lib/objects/active_time_ranges.dart

```diff
@@ -11,26 +11,33 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
-//
+
+import 'package:built_value/built_value.dart';
+import 'package:built_value/serializer.dart';
+part 'active_time_ranges.g.dart';
+
 /// The object holds active time range of a device.
 ///
-/// See document https://developers.google.com/admin-sdk/directory/v1/reference/chromeosdevices#resource
-class ActiveTimeRanges {
-  String _date;
-  String _activeTime;
+/// See document https://developers.google.com/admin-sdk/directory/v1/reference/chromeosdevices#resource.
+
+abstract class ActiveTimeRanges
+    implements Built<ActiveTimeRanges, ActiveTimeRangesBuilder> {
+  /// Anonymus constructor
+  ActiveTimeRanges._();
 
-  /// Constructor with optional variables.
-  ActiveTimeRanges([this._date, this._activeTime]);
+  /// Factory constructor
+  static Serializer<ActiveTimeRanges> get serializer =>
+      _$activeTimeRangesSerializer;
 
-  /// Constructor form a json string.
-  ActiveTimeRanges.fromJson(Map<String, dynamic> json) {
-    _date = json['date'];
-    _activeTime = json['activeTime'];
-  }
+  /// Serializer to parse from Json
+  factory ActiveTimeRanges([void Function(ActiveTimeRangesBuilder) updates]) =
+      _$ActiveTimeRanges;
 
   /// Date of usage of the device.
-  String get date => _date;
+  @nullable
```

Comment:
Are all of these really nullable?

Reply:
Yes, since the  [api](https://developers.google.com/admin-sdk/directory/v1/reference/chromeosdevices#resource) doesn't inform which values will not be received so I put all the variables as optional.